### ISSUE TITLE
Misleading error message "Cannot create datatype" for terminal rule

### ISSUE THREAD

**Issue by kthoms** [eclipse/xtext-core#91]
Given the following grammar which does not inherit from Terminals:

```
grammar org.xtext.example.mydsl.MyDsl

//import "http://www.eclipse.org/emf/2002/Ecore" as ecore

generate myDsl "http://www.xtext.org/example/mydsl/MyDsl"

Model:
    greetings+=Greeting*;

Greeting:
    'Hello' name=ID '!';

terminal ID         : '^'?('a'..'z'|'A'..'Z'|'_') ('a'..'z'|'A'..'Z'|'_'|'0'..'9')*;
```

When the Ecore import is missing the error message is: "Cannot create datatype ID". It is not clear that the actual reason is that EString cannot be resolved. Maybe it is enough to add a hint to add the import? Maybe even add a quickfix?

![screenshot 58](https://cloud.githubusercontent.com/assets/265597/17998671/e0a957c8-6b75-11e6-8bef-7a52b21e09ac.png)

**Comment by miklossy** [comment]
This issue is related to https://bugs.eclipse.org/bugs/show_bug.cgi?id=479137

**Comment by cdietrich** [comment]
fixed in 2.13
